main/ChecksBalancesMod.F90

```diff
@@ -240,7 +243,134 @@ subroutine CheckLitterPools(currentSite,bc_in)
     return
   end subroutine CheckLitterPools
 
+  ! ==================================================================
+
+  subroutine CheckIntegratedMassPools(site)
+
+
+    type(ed_site_type), intent(inout) :: site
+
+    !integer  :: nsites
+    integer  :: el
+    integer  :: s
+    real(r8) :: tot_veg_turnover
+    real(r8) :: tot_litter_input
+    real(r8) :: net_uptake
+    real(r8) :: biomass_stock
+    real(r8) :: seed_stock
+    real(r8) :: litter_stock
+    real(r8) :: total_stock
+
+    logical,parameter :: check_iflux_bal = .true.  
+
+    if(check_iflux_bal) then
+
+       ! For carbon balance checks, we need to initialize the
+       ! total carbon stock
+       do el=1,num_elements
+          call SiteMassStock(site,el,total_stock, &
+               biomass_stock,litter_stock,seed_stock)
+
+          associate(ibal => site%iflux_balance(el), &
+               ediag => site%flux_diags%elem(el), &
+               diag  => site%flux_diags, &
+               site_mass => site%mass_balance(el))
+
+            ! Initialize the integrated flux balance diagnostics
+            ! No need to initialize the instantaneous states, those are re-calculated
+
+            ibal%state_liveveg = (biomass_stock + seed_stock)*area_inv
+            ibal%state_litter  = litter_stock * area_inv
+
+            ! Flux for live veg: net uptake (either NPP or net nutrient uptake) +
+            !                    net spatial seed flux -
+            !                    veg turnover to litter -
+            !                    veg loss to fire (SEEDS DONT BURN) -
+            !                    veg loss from exported harvest - 
+            !                    seed turnover
+
+            tot_litter_input = (sum(ediag%leaf_litter_input(:)) + &
+                 sum(ediag%root_litter_input(:)) + &
+                 sum(ediag%cwd_ag_input(:)) + &
+                 sum(ediag%cwd_bg_input(:)))*area_inv
+
+            select case(element_list(el))
+            case(carbon12_element)
+               net_uptake = diag%npp + site_mass%net_root_uptake*area_inv
+            case(nitrogen_element)
+               net_uptake = site_mass%net_root_uptake*area_inv
+            case(phosphorus_element)
+               net_uptake = site_mass%net_root_uptake*area_inv
+            case default
+               write(fates_log(),*) 'FATES: Invalid choice for cohort_fusion_conservation_method'
```

Comment:
Looks like the message to FATES log needs updating. 'FATES: Invalid element in CheckIntergratedMassPools'

Reply:
fixed